assignments/002-unix-intro/unix-intro.sh

```diff
@@ -0,0 +1,708 @@
+####         Using `gzcat` and a pipe, view the first 8 lines of 
+####         `fastq/DPCh_plate1_C11_S35.R1.fq.gz`
+
+
+zcat fastq/DPCh_plate1_C11_S35.R1.fq.gz | head -n 8
+
+------
+
+@K00364:64:HTYYCBBXX:1:1101:1824:48192/1
+GTAGAATAATAGTGAATCAAATCAAATGTTATTTGTCACATGCGCTGAATACAACAGGTGTGGACCTTACAGTGAAATGCTTCCTTACAAGCCCTTAACCAACAATGCAGTTTTAAGAAAAATGAGTGTGAGATAAGTAAAAAATAGAAAA
++
+AAFFFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJJAJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFJJJJJJJJJJJJ7JJJJJJFFFFJJJJJJ7A7AJJAFAJJJJJFJJJJ-<JFJ-FFFJJJ
+@K00364:64:HTYYCBBXX:1:1101:2869:22080/1
+TTAAAACACGGTATGATGCAAGCAGCACAACACATCAATAACAAAAATACAAGAATTAGGGTCAGAAATCCAGTAACCACCATACTAGTGTACTTACCAAACCAGGCTCCCAACCAAGAGAACAGTCCAGACTCCTCCACCCTCGCCATGG
++
+-AAFFJJJJJJJJJJJJFFJJJFJFJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJJFFJJJ<JJJJJJFFFFJFFJJJJJJAFJJJJJJJJJJJJJJJJJJFAJJJJJJJJJJJJJJJJJJJFJJJFAJJJJJJJFJJJJJJJJFFJJ
+
+#### Ex. 32: Do the same to see the first 8 lines of `fastq/DPCh_plate1_C11_S35.R2.fq.gz`
+
+zcat fastq/DPCh_plate1_C11_S35.R2.fq.gz | head -n 8
+
+-----
+
+@K00364:64:HTYYCBBXX:1:1101:1824:48192/2
+CACAAGGAACGACAGAGGGTAGTGCGTACGGCCCAGCACATCACTGGGACCAAGCATCCTGCAATCCAGGACCTCCATACCAGGCGGTGTCAGAGGAAGGCCCTAAAATTGTCAAAGACTCCAGCCACCCTAGTCATTAACTCTTCGCTCT
++
+A<-A-<-F<AJ-<---<<--<-<-7FFJ-77A<-<--7-<7-<-7<J<77A<--7-AAA7-AF-7FAF<-<FJ---7-77FJA7-77F7FFJ7F<--<--77)A7FA-A7-<---7<<F7-<-7<))7--)<)7<AF----7<<<<-)-7)
+@K00364:64:HTYYCBBXX:1:1101:2869:22080/2
+CTATGATCAACAGCGTTTTGTGATTTACCCCCGTGATGCACTCACTGGTATGTCTGAACAGCTTGAGGCCACATCTAGGGTTGCCAGACAGAATAGACTTGCTTTGGATATGCTTCTTGCCAGTCAGGGGGGTGTCTGTAAGATGTTCGGT
++
+AAAFAFJJJJJJJJJJ-FJJJJJJFJJJJFJJJJJJJJJJJJJFJFFJFFFFFFJFFJJJFJFAAA777AAJJJAA--A<-A<-JJA-<JFJJJ-7<-AAJJFJJJ<JJ<JAFFFF---7<<7777<)7<<-<777-<-7--<FJJ<F-<<
+
+#text switched to green here, not sure why!
+
+#### Ex. 33: What do you notice about the names of the two reads in each 
+####         file (Lines 1 and 5, that start with a `@`)
+
+The names of the two reads are the same: K00364
+
+
+#### Ex. 34: Print the date, redirect it to a file called `now.txt`,
+####         catenate that file to _stdout_ and then remove it
+
+date > now.txt
+
+cat now.txt
+
+-----
+
+Mon Jan 29 11:37:47 MST 2024
+
+rm now.txt
+
+#### Ex. 35: Count the number of lines/words/characters (using `wc`)
+####         in the two SAM files in the `sam` directory
+
+wc sam/DPCh_plate1_A05_S5.sam sam/DPCh_plate1_A06_S6.sam
+
+-----
+
+ 14219  45092 511203 sam/DPCh_plate1_A05_S5.sam
+ 14150  43985 479977 sam/DPCh_plate1_A06_S6.sam
+ 28369  89077 991180 total
+
+
+#### Ex. 36: Note that gzcat will decompress all files you give it and send the 
+####         output to _stdout_. Knowing this, count how many lines are in each of 
+####         the `R1` files in `fastq` (after decompressing them into text files),
+####         and also count up all the lines in the `R1` files in `fastq`. 
+####         Note they should be the same.
+
+for file in fastq/*R1*.fq.gz; do zcat "$file" | wc -l; done
```

Comment:
This is doing each one separately, and wc-ing it separately.  

You can also just do:

```sh
zcat fastq/*R1*.fq.gz
```
and get it all out one one stream